lib/ably/realtime/presence/members_map.rb

```diff
@@ -233,60 +224,31 @@ def setup_event_handlers
           clean_up_members_not_present_after_sync
           change_state :sync_complete
         end
-
-        unsafe_on(:sync_complete) do
-          update_local_member_state
-        end
       end
 
-      # Listen for events that change the PresenceMap state and thus
-      # need to be replicated to the local member set
-      def update_local_member_state
-        new_local_members = members.select do |member_key, member|
-          member.fetch(:message).connection_id == connection.id
-        end.each_with_object({}) do |(member_key, member), hash_object|
-          hash_object[member_key] = member.fetch(:message)
-        end
-
-        @local_members.reject do |member_key, message|
-          new_local_members.keys.include?(member_key)
-        end.each do |member_key, message|
-          re_enter_local_member_missing_from_presence_map message
-        end
-
-        @local_members = new_local_members
-      end
-
-      def re_enter_local_member_missing_from_presence_map(presence_message)
-        local_client_id = presence_message.client_id || client.auth.client_id
-        logger.debug { "#{self.class.name}: Manually re-entering local presence member, client ID: #{local_client_id} with data: #{presence_message.data}" }
-        presence.enter_client(local_client_id, presence_message.data).tap do |deferrable|
-          deferrable.errback do |error|
-            presence_message_client_id = presence_message.client_id || client.auth.client_id
-            re_enter_error = Ably::Models::ErrorInfo.new(
-              message: "unable to automatically re-enter presence channel for client_id '#{presence_message_client_id}'. Source error code #{error.code} and message '#{error.message}'",
-              code: Ably::Exceptions::Codes::UNABLE_TO_AUTOMATICALLY_REENTER_PRESENCE_CHANNEL
-            )
-            channel.emit :update, Ably::Models::ChannelStateChange.new(
-              current: channel.state,
-              previous: channel.state,
-              event: Ably::Realtime::Channel::EVENT(:update),
-              reason: re_enter_error,
-              resumed: true
-            )
+      def enter_local_members
+        local_members.values.each do |member|
+          local_client_id = member.client_id || client.auth.client_id
```

Comment:
Why `|| client.auth.client_id` here? [RTP17g](https://sdk.ably.com/builds/ably/specification/main/features/#RTP17g) just says "using the `clientId` … attributes from that member".

Reply:
Fixed https://github.com/ably/ably-ruby/pull/418/commits/c10e485344ac4f87e7d83445a6ab3b9b287f1129